generate-client.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ -z ${GEN_ROOT:-} ]]; then
```

Comment:
I have a slight preference for the tmp directory because it helps ensure builds are repeatable: you're cloning a fresh copy of the repo and checking out a fresh commit.  it's less likely, for example, that there's random changes or additions that might impact the code generation results.

This is also analogous to how kubernetes-client/gen works (which clones new copies of openapi-generator), so i don't think it's a huge departure that will confuse people.

If you're working on kubernetes-client/gen, you can specify a GEN_ROOT for some local verification testing.  I plan on connecting CI checks you mention in the Issue that assert the code in src/gen is the same code the settings will generate.

Reply:
Yes I agree. I wrote that comment before but you have convinced me.